rust_src/src/buffers.rs

```diff
@@ -129,3 +131,26 @@ pub fn buffer_file_name(buffer: LispObject) -> LispObject {
 
     LispObject::from_raw(buf.filename)
 }
+
+
+#[no_mangle]
+pub extern "C" fn validate_region(b: *mut Lisp_Object, e: *mut Lisp_Object) {
+    let start = LispObject::from_raw(unsafe { *b });
+    let stop = LispObject::from_raw(unsafe { *e });
+
+    let mut beg = start.as_fixnum_coerce_marker_or_error();
```

Comment:
Yes - after some testing this is definitely the problem, but I don't really understand how to fix it.  I was thinking if I used `mem::replace` on the marker with the fixnum at `m.position()` in `as_fixnum_coerce_marker_or_error`, so:
```rust
mem::replace(&mut self, Self::from_fixnum(m.position() as EmacsInt))
``

I get the right return from `as_fixnum_coerce...`, but it doesn't seem to change the value at the original pointer. I thought that since I got the LispObject by converting it from the raw pointer from the input of `validate_region` that should be the right thing to do.

Like I said, I don't seem to be understanding what is going on here.